prefect_transform/credentials.py

```diff
@@ -0,0 +1,77 @@
+"""Transform credentials block"""
+import os
+from typing import Optional
+
+from prefect.blocks.core import Block
+from pydantic import SecretStr, root_validator
+from transform import MQLClient
+from transform.exceptions import AuthException, URLException
+
+from prefect_transform.exceptions import (
+    TransformAuthException,
+    TransformConfigurationException,
+)
+
+
+class TransformCredentials(Block):
+    """
+    Block used to manage authentication with Transform.
```

Comment:
Can we migrate the previous args docstrings here? Also add types so mkdocs works (griffe doesn't work well with pydantic at the moment) and an example?

Here's how prefect-snowflake does it
https://github.com/PrefectHQ/prefect-snowflake/blob/main/prefect_snowflake/credentials.py#L19-L40

The UI will replace `BLOCK_NAME` with the actual block name that the user inputs and show the snippet, like here, I named the block `my-block`
![image](https://user-images.githubusercontent.com/15331990/184705168-3265522d-79b7-4ac6-aeb6-5573b01b9cb1.png)